pkg/handler/crud.go

```diff
@@ -147,10 +147,18 @@ func (c *crud) CreateFlag(params flag.CreateFlagParams) middleware.Responder {
 
 func (c *crud) GetFlag(params flag.GetFlagParams) middleware.Responder {
 	f := &entity.Flag{}
-	err := entity.PreloadSegmentsVariants(getDB()).First(f, params.FlagID).Error
-	if err != nil {
+	result := entity.PreloadSegmentsVariants(getDB()).First(f, params.FlagID)
+
+	// Flag with given ID doesn't exist, so we 404
+	if result.RecordNotFound() {
 		return flag.NewGetFlagDefault(404).WithPayload(
-			ErrorMessage("cannot find flag %v. %s", params.FlagID, err))
+			ErrorMessage("Unable to find flag %v in the database", params.FlagID))
+	}
+
+	// Something else happened, return a 500
+	if  err := result.Error; err != nil {
+		return flag.NewGetFlagDefault(500).WithPayload(
+			ErrorMessage("An unknown error occurred while looking up flag %v: %s", params.FlagID, err))
```

Comment:
```suggestion
			ErrorMessage("an unknown error occurred while looking up flag %v: %s", params.FlagID, err))
```

go prefers starting with a lower case error message